### ISSUE TITLE
Fix S2437 FP: None of the operands is 0

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#6326]
S2437 

### Repro steps

```
public void AfterALoop()
{
    var length = 0x80;
    var bytesRequired = 0;
    while (length > 0)
    {
        length >>= 8;
        bytesRequired++;
    }
    _ = bytesRequired | 0x80; // Noncompliant, FP
}
```

### Expected behavior

The rule should not be raised.

### Actual behavior

The rule is raised for `bytesRequired | 0x80` and none of the operands is `0`.

Source: https://community.sonarsource.com/t/s2437-remove-this-silly-bit-operation-additional-false-positive/74079

**Comment by pavel-mikula-sonarsource** [comment]
Quick guesses: `ConstantValueFinder` -> `CSharpAssignmentFinder` doesn't recognize `++` and considers the left side constant.
